src/app/features/home/details/edit-caption/edit-caption.page.ts

```diff
@@ -52,14 +53,15 @@ export class EditCaptionPage {
       .pipe(
         tap(event => {
           const postMessageEvent = event as MessageEvent;
-
-          if (postMessageEvent.data === 'edit-caption-cancel') {
-            this.navController.back();
-          }
-
-          if (postMessageEvent.data === 'edit-caption-save') {
-            this.iframeService.refreshDetailsPageIframe();
-            this.navController.back();
+          const data = postMessageEvent.data as BubbleToIonicPostMessage;
+          switch (data) {
+            case BubbleToIonicPostMessage.EDIT_CAPTION_CANCEL:
+              this.navController.back();
+              break;
+            case BubbleToIonicPostMessage.EDIT_CAPTION_SAVE:
+              this.iframeService.refreshDetailsPageIframe();
+              this.navController.back();
+              break;
           }
         }),
         untilDestroyed(this)
```

Comment:
**ChatGPT reponse** for ` git show a9e4232dfd700d07b6a61ec73910927941b129ae src/app/features/home/details/edit-caption/edit-caption.page.ts ` 

It looks like a good code refactor that uses enums and switch statements for post message events. It could be further improved by adding error handling in case the received data is not of the expected type, and by using stricter type checking for the BubbleToIonicPostMessage enum, such as adding a union type of string literals for its values.

Reply:
@tammyyang:

For postMessages we don't need to throw error if we got message that we are not interested we just need to skip it so throwing error does not make sense just reacting to events that we are interested is what we need.